storage/s3/src/main/java/io/aiven/kafka/tieredstorage/storage/s3/S3Storage.java

```diff
@@ -46,12 +46,14 @@ public void configure(final Map<String, ?> configs) {
     }
 
     @Override
-    public void upload(final InputStream inputStream, final String key) throws StorageBackendException {
-        try (final var out = s3OutputStream(key)) {
+    public long upload(final InputStream inputStream, final String key) throws StorageBackendException {
+        final S3MultiPartOutputStream out = s3OutputStream(key);
+        try (out) {
```

Comment:
The constructor may throw `AmazonServiceException`, so it still needs to be inside the `try`.